EssentialFeed/EssentialFeedTests/Image Comments/API/ImageCommentsMapperTests.swift

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeed
+
+class ImageCommentsMapperTests: XCTestCase {
+	func test_map_throwsErrorOnNon200HTTPResponse() throws {
+		let json = makeItemsJSON([])
+		let samples = [199, 300, 400, 500]
+
+		try samples.forEach { code in
+			XCTAssertThrowsError(
+				try ImageCommentsMapper.map(json, from: HTTPURLResponse(statusCode: code))
+			)
+		}
+	}
+
+	func test_map_throwsErrorOn200HTTPResponseWithInvalidJSON() throws {
+		let invalidJSON = Data("invalid json".utf8)
+
+		let samples = [199, 300, 400, 500]
+
+		try samples.forEach { sample in
```

Comment:
Suggestion:
Invalid codes can be extracted to a separate getter. It will remove duplication for invalid codes and make tests more readable
```suggestion
		try invalidCodes.forEach { sample in
```

And then in the helpers section
```swift
var invalidCodes: [Int] { return [199, 300, 400, 500] }
```

Reply:
Applied your suggestion.